pkg/products/cloudresources/reconciler.go

```diff
@@ -212,7 +217,17 @@ func (r *Reconciler) Reconcile(ctx context.Context, installation *integreatlyv1a
 		}
 	}
 
-	phase, err = r.reconcileCloudResourceStrategies(client)
+	maintenanceDay, _, err := addon.GetStringParameter(ctx, client, r.ConfigManager.GetOperatorNamespace(), MaintenanceDay)
+	if err != nil {
+		return phase, err
+	}
+
+	maintenanceHour, _, err := addon.GetStringParameter(ctx, client, r.ConfigManager.GetOperatorNamespace(), MaintenanceHour)
+	if err != nil {
+		return phase, err
```

Comment:
```suggestion
		return integreatlyv1alpha1.PhaseFailed, err
	}

	maintenanceHour, _, err := addon.GetStringParameter(ctx, client, r.ConfigManager.GetOperatorNamespace(), MaintenanceHour)
	if err != nil {
		return integreatlyv1alpha1.PhaseFailed, err
```

Let's return `integreatlyv1alpha1.PhaseFailed` here. If we are using the pre-exisiting `phase` from previous function calls, it can be `integreatlyv1alpha1.PhaseCompleted` which would be inaccurate.

Also any thoughts on logging the error ? Alternatively we can move these functions into `reconcileCloudResourceStrategies` instead so that any errors would emit an error event instead 🤔

Reply:
Yes I think it makes sense to move these into `reconcileCloudResourceStrategies` and return errors there where necessary. Pushing changes.